run_local_tests.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+#
+# Copyright (c) 2023 The Flatcar Maintainers.
+# Licensed under the Apache License, Version 2.0 (the "License"); 
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at                         
+#
+# http://www.apache.org/licenses/LICENSE-2.0                  
+#
+# QoL wrapper around ci-automation test.sh for running local tests of qemu_uefi image.
+# The devcontainer tests will be skipped since these require a valid commit ref in
+#   the upstream scripts repo.
+#
+# Requirements:
+# - Docker (for running the Mantle container).
+#
+# Prerequisites:
+# - Flatcar OS image and qemu uefi code to be tested in
+#   __build__/images/images/amd64-usr/latest/
+# 
+#   This script is intended to be run after building a qemu_uefi image with the SDK container:
+#    ./build_packages
+#    ./build_image
+#    ./image_to_vm.sh --from=../build/images/amd64-usr/latest/ --format=qemu_uefi --image_compression_formats none
+#   Then, EXIT the SDK container (or run this on a different terminal):
+#   ./run_local_tests.sh
+#
+# Optional prerequisites:
+# - Custom Mantle container image / version in sdk_container/.repo/manifests/mantle-container.
+#   This comes in handy if you've built a local mantle/kola which you want to test.
+#   Just edit the file and put in the whole containerr image name and version.
+#
+# Output:
+# results reports:
+# - results-qemu_uefi-detailed.md
+# - results-qemu_uefi-detailed.tap
+# - results-qemu_uefi.md
+# - results-qemu_uefi.tap
+# - results-qemu_update-detailed.md
+# - results-qemu_update-detailed.tap
+# - results-qemu_update.md
+# - results-qemu_update.tap
+#
+#
+# - Detailed test run output will reside below __TESTS__/qemu-uefi
+
+function set_vars() {
+  local arch="${1}"
+  local parallel="${2}"
+
+  # Read by the mantle container.
+  # The local directory ("pwd") will be mounted to /work/ in the container.
+  cat > sdk_container/.env <<EOF
+export export QEMU_IMAGE_NAME="/work/__build__/images/images/${arch}-usr/latest/flatcar_production_image.bin"
+export QEMU_UEFI_BIOS="/work/__build__/images/images/${arch}-usr/latest/flatcar_production_qemu_uefi_efi_code.fd"
+export QEMU_UPDATE_PAYLOAD="/work/__build__/images/images/${arch}-usr/latest/flatcar_test_update.gz"
+export PARALLEL_TESTS=${parallel}                                                     
+EOF
+
+  export MAX_RETRIES=5
+  export SKIP_COPY_TO_BINCACHE=1
+}
+#--
+
+function run_local_tests() {
```

Comment:
You are sourcing a file in this function, so I think it would be better to run it in a subshell.
```suggestion
function run_local_tests() (
```